src/with-theme.js

```diff
@@ -0,0 +1,38 @@
+import React, {Component} from 'react'
+
+import {CHANNEL} from './theme-provider'
+import {PropTypes} from './react-compat'
+
+export default function withTheme(ComponentToTheme) {
+  class ThemedComponent extends Component {
+    state = {theme: {}}
+    setTheme = theme => this.setState({theme})
+
+    componentWillMount() {
+      if (this.context[CHANNEL]) {
+        this.setState({theme: this.context[CHANNEL].getState()})
+      }
+    }
+
+    componentDidMount() {
+      if (this.context[CHANNEL]) {
```

Comment:
We actually [discussed](https://github.com/paypal/glamorous/pull/45#discussion_r111036984) adding a warning (and @kkemple even implemented it before I said we should just leave it). Maybe we should add it back after all. I would want it to be wrapped in a `process.env.NODE_ENV === 'production'` call though. I'm uncertain whether our rollup config is currently capable of removing this for our UMD build. May need to investigate that a bit.